glsl/combined.frag

```diff
@@ -64,7 +84,20 @@ vec4 Sample(float samplerIndex, vec2 pos)
 
 void main()
 {
-	vec4 x = Sample(vTexSampler.s, vTexCoord.st);
+	vec2 coords = vTexCoord.st;
+
+	if (EnablePixelArtScaling)
+	{
+		vec2 textureSize = Size(vTexSampler.s);
+		vec2 offset = fract(coords.st * textureSize);
+
+		// Offset the sampling point to simulate bilinear intepolation in window instead of texture coordinates
+		// See https://colececil.io/blog/2017/scaling-pixel-art-without-destroying-it/
+		vec2 interp = clamp(offset * 2 * WindowScale, 0.0, .5) + clamp((offset - 1.0) * 2 * WindowScale + .5, 0.0, .5);
```

Comment:
This is where the problem is: For `WindowScale = 1`, the margin over which we apply the bilinear filter extends to a quarter of the texel width on each side, which results in a blurry image. For `WindowScale = 2`, the margin is only 1/8 of the texel width, so the image is crisper at close zoom levels, but this also makes the aliasing at very far zoom levels worse. 

The margin should instead depend on the actual zoom level and not on `WindowScale` . Replacing with this:
```
vec2 textureSize = Size(vTexSampler.s);
vec2 vUv = coords.st * textureSize;
vec2 alpha = 0.7 * vec2(dFdx(vUv.x), dFdy(vUv.y));		
vec2 offset = fract(vUv);

vec2 interp = clamp(0.5 / alpha * offset, 0.0, .5) + clamp(0.5 / alpha * (offset - 1.0) + .5, 0.0, .5);
coords = (floor(coords.st * textureSize) + interp) / textureSize;
```

solves the problem. See https://csantosbh.wordpress.com/2014/02/05/automatically-detecting-the-texture-filter-threshold-for-pixelated-magnifications/ and https://csantosbh.wordpress.com/2014/01/25/manual-texture-filtering-for-pixelated-games-in-webgl/ (which is where the article in the OP got it's information from in the first place.)

Reply:
Aha, good catch. Add another mark to the "incorrect window size vs viewport size assumption" tally :cry:

We can't assume `GL_OES_standard_derivatives` exists in our GLES 3 support, so I instead pass the pixel-texel ratio in to the shader (replacing the old `EnablePixelScaling` bool).